gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/ApiReactorHandlerFactory.java

```diff
@@ -217,14 +208,34 @@ public ReactorHandler create(Api api) {
         return null;
     }
 
-    protected ExecutionContextFactory executionContextFactory(ComponentProvider componentProvider) {
-        return new ExecutionContextFactory(componentProvider);
+    private boolean isV3ExecutionMode(Api api) {
+        return api.getExecutionMode() == null || api.getExecutionMode() == ExecutionMode.V3;
+    }
+
+    protected <T extends AbstractExecutionContextFactory> T executionContextFactory(
+        Api api,
+        ComponentProvider componentProvider,
+        DefaultReferenceRegister referenceRegister
+    ) {
+        final AbstractExecutionContextFactory executionContextFactory;
+        if (isV3ExecutionMode(api)) {
```

Comment:
I don't really like the fact that we have a condition on execution mode line 160 and then after that you test it again., that means this method is not common on both mode. It make it more confusing.

Reply:
Yes it was more a quick way to avoid duplication of code than make it completely finalized. There is a dedicated task around the `ApiReactorHandlerFactory` that will address the whole problematic around `v3` vs `jupiter` reactor instantiation. I would prefer to leave it like it for now and handle it on the dedicated task later if it's ok for you ?

Reply:
2 distincts methods have been create, one to create a v3 `ExecutionContextFactory`, one for the jupiter `ExecutionContextFactory` (which handle creation of sync or async execution context).